cpp/include/cugraph/graph_functions.hpp

```diff
@@ -903,6 +903,9 @@ weight_t compute_total_edge_weight(
  * @param  select_count The number of vertices to select from the graph
  * @param  with_replacement If true, select with replacement, if false select without replacement
  * @param  sort_vertices If true, return the sorted vertices (in the ascending order).
+ * @param  shuffle_int_to_local If true and If @p given_set is not specified
+ * then shuffle internal (i.e. renumbered) vertices to their local GPUs based on vertex
+ * partitioning, otherwise shuffle as many vertices as local vertex partition size to each GPU.
```

Comment:
I think I have to change all of these. 
Objective: For a graph with N vertices, we want to generate N unique random numbers in range [0, N)  (across GPUs) and we don't want to shuffle the generated numbers to  GPUs based on vertex partitioning, while keeping supporting the existing behavior of `select_random_vertices`.

Reply:
Does it makes sense now @seunghwak ?

Reply:
I guess I'm wondering why this even needs to be part of the `select_random_vertices` call.

While there is randomness to the feature that we need, I think all that we need is a "distributed shuffle" function.  Given an `rmm::device_uvector<T>` on each GPU that is initialized with some values, we want the equivalent of `thrust::shuffle` of that device vector.

I'm not aware of an algorithm that would need to both select a random subset of the vertex list and randomly order that list across all of the GPUs.  If there was such an algorithm it could first call `select_random_vertices` and then call a new `randomly shuffle a distributed device vector` function.

I don't see a benefit of conflating the two actions into a single function.

Reply:
This changes will be aborted. Shuffling will be done by the ECG implementation itself.

Reply:
If this PR is not meant to be merged, should we better close this PR?

Reply:
Actually all the changes to the select_random_vertices has been aborted, but the test has been updated. Please review the test.